crate_universe/src/cli/splice.rs

```diff
@@ -147,5 +152,47 @@ pub fn splice(opt: SpliceOptions) -> Result<()> {
     std::fs::copy(cargo_lockfile_path, output_dir.join("Cargo.lock"))
         .context("Failed to copy lockfile")?;
 
+    if let SplicerKind::Workspace { path, .. } = prepared_splicer {
+        println!("MANIFEST {}", path);
+
+        let child = cargo
+            .command()?
+            .env("RUSTC_WRAPPER", opt.rustc)
+            .env("CARGO_CACHE_RUSTC_INFO", "0")
+            .current_dir(path.parent().expect("All manifests should have a valid parent."))
+            .arg("metadata")
+            .arg("--no-deps")
+            .stdout(std::process::Stdio::piped())
+            .stderr(std::process::Stdio::piped())
+            .spawn()
+            .with_context(|| {
+                format!(
+                    "Error spawning cargo in child process to compute crate paths for workspace '{}'",
+                    path
+                )
+            })?;
+
+        let output = child.wait_with_output().with_context(|| {
+            format!(
+                "Error running `cargo metadata --no-deps` manifest path '{}'",
+                path
+            )
+        })?;
+        if !output.status.success() {
+            tracing::error!("{}", String::from_utf8_lossy(&output.stdout));
+            tracing::error!("{}", String::from_utf8_lossy(&output.stderr));
+            bail!(format!("Failed to run cargo metadata: {}", output.status))
+        }
+
+        let metadata: CargoMetadata =
+            serde_json::from_str(&String::from_utf8_lossy(&output.stdout))
+                .context("Unable to deserialize Cargo metadata")?;
```

Comment:
Is there are reason not to:
```suggestion
        let metadata = cargo.metadata_command_with_options(path, vec![String::from("--no-deps")]).exec().with_context(...)?;`
```

Reply:
If the env vars are useful, let's try to set them everywhere :) `Cargo` already has a `rustc_path` field, so we should be able to unconditionally set `RUSTC_WRAPPER` if needed. I've not seen `CARGO_CACHE_RUSTC_INFO` before, but if it's worth setting for this call, it's presumably worth setting for others too?